tests/integration/ardrive.int.test.ts

```diff
@@ -132,150 +135,281 @@ describe('ArDrive class - integrated', () => {
 		});
 
 		describe('movePublicFolder', () => {
-			beforeEach(() => {
-				stub(arfsDao, 'getPublicDrive').resolves(stubPublicDrive);
-				stub(arfsDao, 'getPublicFolder').resolves(stubPublicFolder);
-				stub(arfsDao, 'getPublicChildrenFolderIds').resolves([stubEntityIDAltTwo]); // Uses Alt ID Two
-			});
-
+			const folderHierarchy = {
+				rootFolder: stubPublicFolder(stubEntityIDRoot, 'root folder'),
+				parentFolder: stubPublicFolder(stubEntityIDParent, stubEntityIDParent),
+				childFolder: stubPublicFolder(stubEntityIDChild, stubEntityIDParent),
+				grandChildFolder: stubPublicFolder(stubEntityIDGrandchild, stubEntityIDChild)
+			};
 			it('throws an error if its being moved inside any of its children folders', async () => {
+				stub(arfsDao, 'getPublicFolder').resolves(folderHierarchy.rootFolder);
+				stub(arDrive, 'getDriveIdAndAssertDrive').resolves(stubEntityID);
+				stub(arfsDao, 'getPublicChildrenFolderIds').resolves([
+					folderHierarchy.parentFolder.entityId,
+					folderHierarchy.childFolder.entityId,
+					folderHierarchy.grandChildFolder.entityId
+				]);
+
 				await expectAsyncErrorThrow({
 					promiseToError: arDrive.movePublicFolder({
-						folderId: stubEntityID,
-						newParentFolderId: stubEntityIDAltTwo // Alt ID Two will be in Children folder ids
+						folderId: folderHierarchy.rootFolder.entityId,
```

Comment:
I would have expected this to throw for being a root folder that got moved, NOT because of the child relationship.

Reply:
07acb87